examples/animation/animate_decay.py

```diff
@@ -1,17 +1,29 @@
+"""
+Animated plot showing a decay process with updating x-scale.
+"""
+
 import numpy as np
 import matplotlib.pyplot as plt
 import matplotlib.animation as animation
 
 
 def data_gen(t=0):
-    cnt = 0
-    while cnt < 1000:
-        cnt += 1
+    """Generate data
+
+    Generates a decaying sine wave.
+    """
+    count = 0
+    while count < 1000:
```

Comment:
Can you change this to a `for count in range(1000):`  That is slightly more pythonic.